### ISSUE TITLE
Auto update doesn't work

### ISSUE THREAD

**Issue by alesju** [AugurTech/handset-detect#10]
Hi, attempting to set autoUpdate to true and verbose to true yields nothing. I even tried adding debug statements to the files and nothing shows up.

I changed the manageDB.js file to instantly perform a downloadUpdatedDB() call to ensure it DOES work as we are now using the enterprise database. No go.

Any idea what could be causing this? I also noticed one of the URLS returns a 301 error and the other returns a different database than the one I can manually download.

**Comment by NawarA** [comment]
Do you have an API key and other information like that needed to enable the auto update to download from the third party service?

**Comment by alesju** [comment]
Yes. We have a ultimate API key and such that we are paying for.

Another thing I noticed, is there a way to expose more information about 
the device? Almost all phones in this module resolves as generic android 
phone without even a brand whereas the official libraries are somewhat 
overkill, it would be nice to know brand and model (where available).

Nawar Alsafar wrote:

**Comment by NawarA** [comment]
I can check go make sure the brand and model are included. I believe an older version of this library had that, and then it was accidentally stripped out the last sem major update. Let's definitely add it back in! Also, if you increase your file descriptor limits to max that the OS allows, does that allow for auto update to work?

**Comment by alesju** [comment]
I replaced fs with graceful-fs and there was no need to up the limit (it's a drop in replacement). However the url this calls is not the correct one it seems. At least it is different from the one found on handsetdetection.

**Comment by NawarA** [comment]
That's awesome! Should we use `graceful-fs` to create databases then vs normal `fs`?

Also, here's the endpoint we use for autoUpdating the database 
`http://api.handsetdetection.com/apiv4/device/fetcharchive.json`

Is there another one you're referring to?

**Comment by alesju** [comment]
I think you should use graceful-fs if you confirm everything works like before, it saves everyone the trouble of increasing the max open files. 

As for the archive I'll have to get back to you with the address we were given.

**Comment by NawarA** [comment]
I added Richard Uren, the creator, to the repo. Hopefully he'll get it resolved. 

Meanwhile, I'm adding `graceful-fs` and `aliases` back to the repo

**Comment by NawarA** [comment]
Just published version `8.1.0` which includes `graceful-fs` and `aliases`. Hopefully Richard will contribute too :)

**Comment by alesju** [comment]
Thanks, this seems to be the URL we get our ultimate archive from.
https://app.handsetdetection.com/sites/getArchive4.json

**Comment by elricho** [comment]
Hey All, https://app.handsetdetection.com/sites/getArchive4.json is the manual download URL, for use inside the control panel. It downloads the same file that the api kits fetch, just via a different mechanism. For automating downloads its best to use the API with  http://api.handsetdetection.com/apiv4/device/fetcharchive.json

Full docs at : https://handsetdetection.readme.io/docs/fetcharchive
